uelc/main/views.py

```diff
@@ -555,33 +557,91 @@ def post(self, request):
         print action_args
-        return HttpResponseRedirect('/uelcadmin/')
 
+        url = request.META['HTTP_REFERER']
+        return HttpResponseRedirect(url)
 
-class UELCAdminHierarchyView(LoggedInMixinSuperuser,
-                             TemplateView):
-    template_name = "pagetree/uelc_admin.html"
-    extra_context = dict()
 
-    def post(self, request):
-        name = request.POST.get('name', '')
-        url = '/pages/'+request.POST.get('url', '')+'/'
-        hier = Hierarchy.objects.filter(Q(base_url=url) | Q(name=name))
+class UELCAdminCreateHierarchyView(LoggedInMixinSuperuser,
+                                   TemplateView):
+        template_name = "pagetree/uelc_admin.html"
+        extra_context = dict()
+
+        def post(self, request):
+            name = request.POST.get('name', '')
+            url = '/pages/' + name + '/'
+            hier = Hierarchy.objects.filter(Q(base_url=url) | Q(name=name))
 
-        if len(hier) > 0:
+            if len(hier) > 0:
+                action_args = dict(
+                    error="Hierarchy exists! Please use the exisiting one,\
+                          or create one with a different name and url.")
+                return action_args
+
+            hier = Hierarchy.objects.create(
+                base_url=url,
+                name=name)
+            hier.save()
             action_args = dict(
-                error="Hierarchy exists! Please use the exisiting one,\
-                      or create one with a different name and url.")
-            return action_args
+                name=hier.name, value=hier.pk, url=hier.base_url, error=None)
+            url = request.META['HTTP_REFERER']
+            return HttpResponseRedirect(url)
 
-        hier = Hierarchy.objects.create(
-            base_url=url,
-            name=name)
-        hier.save()
+
+class UELCAdminDeleteHierarchyView(LoggedInMixinSuperuser,
+                                   TemplateView):
+    extra_context = dict()
+
+    def post(self, request):
+        hierarchy_id = request.POST.get('hierarchy_id')
+        hier = Hierarchy.objects.get(id=hierarchy_id)
+        hier.delete()
         action_args = dict(
-            name=hier.name, value=hier.pk, url=hier.base_url, error=None)
+            hierarchy_id=hierarchy_id,
+            error=None)
         print action_args
-        return HttpResponseRedirect('/uelcadmin/')
+        url = request.META['HTTP_REFERER']
+        return HttpResponseRedirect(url)
+
+
+class UELCAdminCaseView(LoggedInMixinSuperuser,
+                        TemplateView):
+    template_name = "pagetree/uelc_admin_case.html"
+    extra_context = dict()
+
+    def dispatch(self, request, *args, **kwargs):
+        return super(UELCAdminCaseView, self).dispatch(
+            request, *args, **kwargs)
+
+    def get_case_from_hierarchy(self, hierarchy):
+        case = hierarchy.case_set.all()
+        if case:
+            case = case[0]
+        return case
```

Comment:
I think that this method could just be replaced with `return hierarchy.case_set.first()`